docs/easyconfig-files-local-variables.md

```diff
@@ -1,181 +1,166 @@
-  see :ref:`easyconfig_files_local_variables_strict_naming`
-
-* using ``eb --fix-deprecated-easyconfigs``, the names of local variables can be changed automatically to match
-  the recommended naming scheme; see :ref:`easyconfig_files_local_variables_fix_deprecated_easyconfigs`
+For more general information on writing easyconfig files, please see
+[Writing easyconfig files: the basics][writing_easyconfig_files].
+
+## Motivation & context {: #easyconfig_files_local_variables_motivation }
+
+When composing easyconfig files, it can sometimes make sense to use one
+or more *local variables* to define easyconfig parameters in a cleaner
+way, for example to avoid copy-pasting values that are used multiple
+times, or to avoid too long lines (longer than the maximum width of 120
+characters as specified in [Code style][code_style]).
+A local variable within the context of easyconfig files is any variable
+that does not correspond to a known easyconfig parameter (either general
+of easyblock-specific).
+
+One thing that is easily overlooked is the importance of how these local
+variables are named: you should try and avoid that the names of local
+variables will match with the name of an easyconfig parameter that may
+get introduced in future EasyBuild versions. If that would happen, the
+semantics of the easyconfig file will change which may either result in
+a broken installation or perhaps a different installation than was
+intended (which could be worse than a broken one).
+
+Avoiding that names of local variables are *close* to the name of
+(future) easyconfig parameters is also important, because of the "typo
+detection" feature that EasyBuild has: if the name of a local variable
+is too close to a known easyconfig parameter, EasyBuild will assume you
+have made a mistake, and will report a typo error, for example:
+
+``` console
+ERROR: Failed to process easyconfig example.eb:
+You may have some typos in your easyconfig file: configopt -> configopts
+```
 
+## Changes in EasyBuild v4.0 w.r.t. local variables in easyconfig files {: #easyconfig_files_local_variables_easybuild4 }
 
+Some changes were made in EasyBuild v4.0 to discourage the use of poorly
+named local variables:
 
-.. _easyconfig_files_local_variables_naming_scheme:
+- a [Recommended naming scheme for local variables in easyconfig files][easyconfig_files_local_variables_naming_scheme] was defined;
+- a mechanism was implemented to detect the use of local variables in
+    easyconfig files, which will print a
+    [Warning for local variables that do not follow the recommended naming scheme][easyconfig_files_local_variables_naming_warning]
+- EasyBuild can be configured to report an error for local variables
+    that do not follow the recommended naming scheme; see
+    [Specifying what should be done when non-confirming local variables are found via `--local-var-naming-check`][easyconfig_files_local_variables_check_mode]
```

Comment:
The original link here `easyconfig_files_local_variables_strict_naming` was a dead-link so I took this as the closest match.